code/events/test_events.py

```diff
@@ -0,0 +1,170 @@
+                    line_nbr = (
+                        file.count("\n", 0, start) + 1
+                    )  # Calculate the corresponding line number
+
+                    # Search for the capture pattern within seven lines before the match
+                    previous_lines = file.split("\n")[line_nbr - 7 : line_nbr]
+                    previous_lines_text = "\n".join(previous_lines)
+
+                    fix_pos_pattern = r"([\d.]+)\s+EXP\s+New trial \(rep=\d+, index=\d+\): OrderedDict\(\[\(\'xpos\', (-?\d+\.\d+)\), \(\'ypos\', (-?\d+\.\d+)\)\]\)"
+
+                    # Find all the matches using the regular expression
+                    matches = re.findall(fix_pos_pattern, previous_lines_text)
+                    # There should be only one match, otherwise we risk mixing up different instances
+                    assert len(matches) == 1
+                    timestamp, xpos, ypos = matches[0]
+                    value = f"[{xpos}, {ypos}]"
+
+                # If no trigger were recorded in the psychopy log, we need to approximate its timestamp
+                # with the closest log event.
+                if not timestamp:
+                    if keyword == "movie":
+                        # We coded the resting-state task such that the movie starts at the trigger.
+                        trigger_timestamp = onset
+                    elif keyword in ["blank", "cog", "mot", "cog"]:
+                        # The closest event for qct is "EXP 	eyetracker.clearEvents()"
+                        trigger_pattern = r"(\d+\.\d+)\s+EXP\s+eyetracker.clearEvents()"
+                        trigger_timestamp = float(re.findall(trigger_pattern, file)[0])
+                    else:
+                        # The closest event for bht is "EXP  text_2: autoDraw = False"
+                        trigger_pattern = r"(\d+\.\d+)\s+EXP\s+text_2: autoDraw = False"
+                        trigger_timestamp = float(re.findall(trigger_pattern, file)[0])
+
+                # Subtract the timestamp of the first trigger to the onset of the task to get events
+                # onset in the fMRI recording time.
+                onset = onset - trigger_timestamp
+
+                # Keep only 1 decimal of precision
+                onset = round(onset, 1)
+                duration = round(duration, 1)
+
+                # We have all the information needed for the event, it can be inserted in the dataframe.
+                event = {
+                    "onset": onset,
+                    "duration": duration,
+                    "trial-type": trial_type,
+                    "value": value,
+                }
+                event_dataframe = pd.concat(
+                    [event_dataframe, pd.DataFrame([event])], ignore_index=True
+                )
+
+                # Remove the start timestamp from the dictionary to avoid double counting
+                del start_timestamp[keyword]
```

Comment:
Probably best to remove with a pop above. Otherwise, you might not execute this line and then nothing works.
```suggestion
```